response_operations_ui/views/reporting_units.py

```diff
@@ -87,19 +87,47 @@ def edit_contact_details(ru_ref, respondent_id):
     edit_details_data = {
         "first_name": request.form.get('first_name'),
         "last_name": request.form.get('last_name'),
-        "email": request.form.get('email'),
+        "email_address": request.form.get('hidden_email'),
+        "new_email_address": request.form.get('email'),
         "telephone": request.form.get('telephone'),
         "respondent_id": respondent_id}
 
-    edit_successfully = edit_contact_details_controller.edit_contact_details(edit_details_data, respondent_id)
+    respondent_details = edit_contact_details_controller.get_contact_details(respondent_id)
+
+    edit_successfully, error_type = edit_contact_details_controller.edit_contact_details(edit_details_data,
+                                                                                         respondent_id)
 
     if not edit_successfully:
-        respondent_details = edit_contact_details_controller.get_contact_details(respondent_id)
-        logger.info('Error submitting respondent details', respondent_id=respondent_id)
-        return render_template('edit-contact-details.html', ru_ref=ru_ref, form=form, error=True,
-                               respondent_details=respondent_details)
 
-    return redirect(url_for('reporting_unit_bp.view_reporting_unit', ru_ref=ru_ref, edit_details=True))
+        logger.info('Error submitting respondent details', respondent_id=respondent_id)
+        if error_type == 'bad-email':
+            return render_template('edit-contact-details.html', ru_ref=ru_ref, form=form, email_error=True,
+                                   respondent_details=respondent_details)
+        else:
+            return render_template('edit-contact-details.html', ru_ref=ru_ref, form=form, error=True,
+                                   respondent_details=respondent_details)
+
+    message = None
+    details_changed = False
+    email_changed = False
+
```

Comment:
These 'ifs' feels like they should be in the controller, and the controller should just return what has changed